pkg/cmd/admin/registry/registry.go

```diff
@@ -176,6 +177,13 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		return fmt.Errorf("unable to configure printer: %v", err)
 	}
 
+	// Check if the specified service account already exists
+	_, err = kClient.ServiceAccounts(namespace).Get(cfg.ServiceAccount)
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("can't check for existing service-account %q: %v", cfg.ServiceAccount, err)
+	}
+	generateServiceAccount := err != nil
```

Comment:
if it finds the user, isn't ```err == nil``` so that ```generateServiceAccount``` becomes ``False``?

At this point ```err``` is set only when ```errors.IsNotFound(err)``` as otherwise would exit earlier.